be/src/vec/exec/format/parquet/vparquet_reader.cpp

```diff
@@ -355,25 +438,34 @@ Status ParquetReader::_init_row_group_readers() {
     return Status::OK();
 }
 
-Status ParquetReader::_filter_row_groups() {
-    if (_total_groups == 0 || _t_metadata->num_rows == 0 || _range_size < 0) {
+Status ParquetReader::_filter_row_groups(const bool& enabled,
+                                         std::vector<RowGroupIndex>& group_indexes) {
+    if (enabled && (_total_groups == 0 || _t_metadata->num_rows == 0 || _range_size < 0)) {
         return Status::EndOfFile("No row group need read");
     }
+    int32_t start_row_id = 0;
```

Comment:
if `enabled` is false, you can return immediately.

Reply:
whether it is enabled or not，we will generate group index for read in parallel or other optimizations